### ISSUE TITLE
Fix false positives for ::part in selector-pseudo-element-no-unknown

### ISSUE THREAD

**Issue by bennypowers** [stylelint/stylelint#4589]
> What is the problem you're trying to solve?

Using CSS Shadow Parts violates linter rules `selector-pseudo-element-no-unknown` for the `::part` pseudo element and `selector-type-no-unknown` for the argument to `::part()`

> What solution would you like to see?

I would like to see the`selector-pseudo-element-no-unknown` rule updated to allow CSS Shadow Part selection.
```css
stripe-elements::part(stripe) {
  border: none;
  box-shadow: none;
}
```

A plugin that scans `custom-elements.json` files for a all custom-elements imported by the project and validates the names of their Shadow Parts would be amazing, but is perhaps out of scope of this request.

**Comment by evilebottnawi** [comment]
Feel free to send a PR with new pseudo elements, you can name then `shadowTreePseudoElements` https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js#L205

**Comment by jeddy3** [comment]
@bennypowers Thanks for the request and for using the template.

I've labeled the issue as ready to implement. As @evilebottnawi said, please consider [contributing](https://github.com/stylelint/stylelint/blob/master/CONTRIBUTING.md) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

> A plugin that scans custom-elements.json files for a all custom-elements imported by the project and validates the names of their Shadow Parts would be amazing

Yes, this is out of scope for a built-in rule but would make for an excellent [plugin](https://stylelint.io/developer-guide/plugins).
